idx/metric_definition.go

```diff
@@ -354,9 +355,13 @@ func (md *MetricDefinition) NameWithTags() string {
 	return bld.String()
 }
 
-func (md *MetricDefinition) NameWithTagsHash() string {
-	buffer := bytes.NewBufferString(md.NameWithTags())
-	return fmt.Sprintf("%x", md5.Sum(buffer.Bytes()))
+func (md *MetricDefinition) NameWithTagsHash() Md5Hash {
+	md5Sum := md5.Sum(bytes.NewBufferString(md.NameWithTags()).Bytes())
+	ret := Md5Hash{
+		Upper: binary.LittleEndian.Uint64(md5Sum[:8]),
+		Lower: binary.LittleEndian.Uint64(md5Sum[8:]),
+	}
+	return ret
```

Comment:
I'd add some function `idx.Md5HashFromString(string) Md5Hash`, since this snippet of specific and repetitive

Reply:
Not sure I follow how it is repetitive. I feel like it is more closely tied to the MetricDefinition than anything else. It only appears in one more place which is in memory.go.

Reply:
i can see both sides of the argument. let's just leave as is and move on